servermaster/jobmanager.go

```diff
@@ -69,17 +70,21 @@ func (jm *JobManagerImplV2) SubmitJob(ctx context.Context, req *pb.SubmitJobRequ
 		resp.Err = errors.ToPBError(err)
 		return resp
 	}
+
+	// TODO: data persistence for masterConfig
+
 	// CreateWorker here is to create job master actually
 	// TODO: use correct worker type and worker cost
 	id, err := jm.BaseMaster.CreateWorker(
```

Comment:
`CreateWorker` can fail at last, because the rpc is done in a background goroutine. Is this situation well handled here?

Reply:
added a state transition when meeting error in `OnWorkerDispatched`, from `WaitAck` to `Pending`.